### ISSUE TITLE
Cloud Module - cloudtrail - Error on existing trail without S3 prefix

### ISSUE THREAD

**Issue by mdaize** [ansible/ansible-modules-extras#2166]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

AWS - Cloudtrail
##### ANSIBLE VERSION

```
ansible 2.0.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### SUMMARY

Describing (view) existing trails throws a module failure due to a missing key in the returned dictionary.
##### STEPS TO REPRODUCE
- Create a cloudtrail using existing module without a S3 prefix
- Re-run using existing cloudtrail configuration -- again, without s3 prefix
##### EXPECTED RESULTS

Nothing (unless there are changes)
##### ACTUAL RESULTS

Module error occurs due to KeyError as S3KeyPrefix is NOT part of the results when a describe-trail is performed.

```
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/home/<username>/.ansible/tmp/ansible-tmp-1462399000.22-104944138960760/cloudtrail", line 2435, in <module>
    main()
  File "/home/<username>/.ansible/tmp/ansible-tmp-1462399000.22-104944138960760/cloudtrail", line 194, in main
    results['view']['S3KeyPrefix']                != s3_key_prefix  or \
KeyError: 'S3KeyPrefix'
```

You can see the key does not exist if you perform `aws cloudtrail describe-trails`:

```
        {
            "IncludeGlobalServiceEvents": true, 
            "Name": "<trail-name>", 
            "TrailARN": "arn:aws:cloudtrail:ap-southeast-2:account_id:trail/<trail-name>", 
            "LogFileValidationEnabled": true, 
            "IsMultiRegionTrail": true, 
            "S3BucketName": "<s3-bucket-name>", 
            "HomeRegion": "ap-southeast-2"
        }
```

even though the code is expecting it as seen here on line 193 of `ansible-modules-extras/cloud/amazon/cloudtrail.py`:

```
              results['view']['S3KeyPrefix']                != s3_key_prefix  or \
```

**Comment by ansibot** [comment]
@ansible ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by resmo** [comment]
resolved_by_pr #2939
